### ISSUE TITLE
Build scrcpy error

### ISSUE THREAD

**Issue by zhanweisun** [Genymobile/scrcpy#785]
build scrcpy error 
log 
```
[39/39] Generating scrcpy-server with a custom command.
FAILED: server/scrcpy-server.jar 
/home/jack/Dev/Code/scrcpy/server/./scripts/build-wrapper.sh /home/jack/Dev/Code/scrcpy/server server/scrcpy-server.jar release
Starting a Gradle Daemon (subsequent builds will be faster)
The message received from the daemon indicates that the daemon has disappeared.
Build request sent: Build{id=f48f54ae-7189-41b1-8b92-1f4504b26eb9, currentDir=/home/jack/Dev/Code/scrcpy/x}
Attempting to read last messages from the daemon log...
Daemon pid: 12273
  log file: /home/jack/.gradle/daemon/5.5.1/daemon-12273.out.log
----- Last  20 lines from daemon log file - daemon-12273.out.log -----
11:41:38.331 [ERROR] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] Command execution: started DaemonCommandExecution[command = Build{id=f48f54ae-7189-41b1-8b92-1f4504b26eb9, currentDir=/home/jack/Dev/Code/scrcpy/x}, connection = DefaultDaemonConnection: socket connection from /0:0:0:0:0:0:0:1:34235 to /0:0:0:0:0:0:0:1:44346] after 0.0 minutes of idle
11:41:38.332 [INFO] [org.gradle.launcher.daemon.server.DaemonRegistryUpdater] Marking the daemon as busy, address: [4ca9c75a-bf1c-4b85-9369-23e44f9a00a5 port:34235, addresses:[/0:0:0:0:0:0:0:1%lo, /127.0.0.1]]
11:41:38.332 [DEBUG] [org.gradle.launcher.daemon.registry.PersistentDaemonRegistry] Marking busy by address: [4ca9c75a-bf1c-4b85-9369-23e44f9a00a5 port:34235, addresses:[/0:0:0:0:0:0:0:1%lo, /127.0.0.1]]
11:41:38.334 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Waiting to acquire exclusive lock on daemon addresses registry.
11:41:38.334 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Lock acquired on daemon addresses registry.
11:41:38.336 [DEBUG] [org.gradle.cache.internal.DefaultFileLockManager] Releasing lock on daemon addresses registry.
11:41:38.337 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] resetting idle timer
11:41:38.337 [DEBUG] [org.gradle.launcher.daemon.server.DaemonStateCoordinator] daemon is running. Sleeping until state changes.
11:41:38.338 [INFO] [org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy] Daemon is about to start building Build{id=f48f54ae-7189-41b1-8b92-1f4504b26eb9, currentDir=/home/jack/Dev/Code/scrcpy/x}. Dispatching build started information...
11:41:38.339 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 19: dispatching class org.gradle.launcher.daemon.protocol.BuildStarted
11:41:38.342 [DEBUG] [org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment] Configuring env variables: {PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin, XAUTHORITY=/home/jack/.Xauthority, INVOCATION_ID=cfba65df218f4eca8b46bcfd1d1b98a2, XMODIFIERS=@im=fcitx, XDG_DATA_DIRS=/usr/share/budgie-desktop:/usr/local/share:/usr/share:/var/lib/snapd/desktop, GDMSESSION=budgie-desktop, MANDATORY_PATH=/usr/share/gconf/budgie-desktop.mandatory.path, TEXTDOMAINDIR=/usr/share/locale/, GTK_IM_MODULE=fcitx, DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/1000/bus,guid=7940d85b6c72d41a407cce8f5d705c5e, ANDROID_HOME=/home/jack/Dev/AndroidSDK/, DEFAULTS_PATH=/usr/share/gconf/budgie-desktop.default.path, XDG_CURRENT_DESKTOP=Budgie:GNOME, JOURNAL_STREAM=9:33247, SSH_AGENT_PID=1597, COLORTERM=truecolor, QT4_IM_MODULE=fcitx, SESSION_MANAGER=local/Jack-P150HMx:@/tmp/.ICE-unix/1511,unix/Jack-P150HMx:/tmp/.ICE-unix/1511, LOGNAME=jack, PWD=/home/jack/Dev/Code/scrcpy/x, MANAGERPID=1492, IM_CONFIG_PHASE=2, LANGUAGE=zh_CN:zh, SHELL=/bin/bash, LESSOPEN=| /usr/bin/lesspipe %s, OLDPWD=/home/jack/Dev/Code/scrcpy, GNOME_DESKTOP_SESSION_ID=this-is-deprecated, GTK_MODULES=gail:atk-bridge, CLUTTER_IM_MODULE=xim, XDG_SESSION_PATH=/org/freedesktop/DisplayManager/Session0, TEXTDOMAIN=im-config, DBUS_STARTER_ADDRESS=unix:path=/run/user/1000/bus,guid=7940d85b6c72d41a407cce8f5d705c5e, XDG_SESSION_DESKTOP=budgie-desktop, LS_COLORS=rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.Z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:, SHLVL=2, LESSCLOSE=/usr/bin/lesspipe %s %s, QT_IM_MODULE=fcitx, TERM=xterm-256color, XDG_CONFIG_DIRS=/etc/xdg/xdg-budgie-desktop:/etc/xdg, LANG=zh_CN.UTF-8, XDG_SEAT_PATH=/org/freedesktop/DisplayManager/Seat0, XDG_SESSION_ID=c2, XDG_SESSION_TYPE=x11, DISPLAY=:0, QT_STYLE_OVERRIDE=, _=/home/jack/Dev/Code/scrcpy/server/../gradlew, GDM_LANG=zh_CN, XDG_GREETER_DATA_DIR=/var/lib/lightdm-data/jack, GPG_AGENT_INFO=/run/user/1000/gnupg/S.gpg-agent:0:1, DESKTOP_SESSION=budgie-desktop, USER=jack, XDG_MENU_PREFIX=gnome-, VTE_VERSION=5201, QT_ACCESSIBILITY=1, SSH_AUTH_SOCK=/run/user/1000/keyring/ssh, XDG_SEAT=seat0, QT_QPA_PLATFORMTHEME=gtk2, XDG_VTNR=7, TILIX_ID=4a68f477-d5ea-484a-9277-9b8b61982ab1, XDG_RUNTIME_DIR=/run/user/1000, DBUS_STARTER_BUS_TYPE=session, HOME=/home/jack}
11:41:38.344 [DEBUG] [org.gradle.launcher.daemon.server.SynchronizedDispatchConnection] thread 17: received class org.gradle.launcher.daemon.protocol.CloseInput
11:41:38.344 [DEBUG] [org.gradle.launcher.daemon.server.DefaultDaemonConnection] thread 17: Received IO message from client: org.gradle.launcher.daemon.protocol.CloseInput@29edbd4a
11:41:38.348 [DEBUG] [org.gradle.launcher.daemon.server.exec.LogToClient] About to start relaying all logs to the client via the connection.
11:41:38.348 [INFO] [org.gradle.launcher.daemon.server.exec.LogToClient] The client will now receive all logging from the daemon (pid: 12273). The daemon log file: /home/jack/.gradle/daemon/5.5.1/daemon-12273.out.log
11:41:38.350 [INFO] [org.gradle.launcher.daemon.server.exec.LogAndCheckHealth] Starting build in new daemon [memory: 1.4 GB]
11:41:38.351 [INFO] [org.gradle.launcher.daemon.server.exec.ForwardClientInput] Closing daemon's stdin at end of input.
11:41:38.351 [INFO] [org.gradle.launcher.daemon.server.exec.ForwardClientInput] The daemon will no longer process any standard input.
11:41:38.357 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] The daemon has started executing the build.
11:41:38.357 [DEBUG] [org.gradle.launcher.daemon.server.exec.ExecuteBuild] Executing build with daemon context: DefaultDaemonContext[uid=bf7b6c6e-f812-4afb-ac00-4fc532f45c39,javaHome=/usr/lib/jvm/java-8-openjdk-amd64,daemonRegistryDir=/home/jack/.gradle/daemon,pid=12273,idleTimeout=10800000,priority=NORMAL,daemonOpts=-Xmx1536m,-Dfile.encoding=UTF-8,-Duser.country=CN,-Duser.language=zh,-Duser.variant]
----- End of the daemon log -----


FAILURE: Build failed with an exception.

* What went wrong:
Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
```

**Comment by rom1v** [comment]
> Gradle build daemon disappeared unexpectedly (it may have been killed or may have crashed)

A problem with your `gradle`.

You could try with the [prebuilt server](https://github.com/Genymobile/scrcpy/blob/master/BUILD.md#prebuilt-server) if you just want the last release.

**Comment by tukusejssirs** [comment]
I have experienced a similar error. My issue was that I had not installed the [Android SDK Tools](https://developer.android.com/studio/index.html). I have installed in using [this script](https://gitlab.com/tukusejssirs/lnx_scripts/-/blob/b97182d6b17a59083ae18ca911ce9085005b6736/bash_progs/android_sdk_tools.sh).

Then, I needed to export the `ANDROID_HOME` path, but I have found the `export ANDROID_HOME=~/android/sdk` command in the [BUILD.md](https://github.com/Genymobile/scrcpy/blob/master/BUILD.md#prebuilt-server) quite ambiguous—should I export the `tools` folder location or the `tools/bin` folder location?

I also wanted to change the `--sdk_root` path for installation of the `platform-tools`, but when I did so, `ninja` command failed with similar error than the one stated in the OP. Therefore, ATM we must install the Android SDK Tools without `--sdk_root` option specified. One does not even need to manually install the `platform-tools`, as `ninja` command installs it if not installed.

So, the `ANDROID_HOME` should contain the parent folder of `tools` (SDK files). As I have installed Android SDK Tools in `/opt/android_sdk_tools`, I had to export the following:

```bash
export ANDROID_HOME='/opt/android_sdk_tools'
```

Just to make it clear for everyone, here’s the list of the `ANDROID_HOME` folder contents (after `ninja` command was issued):

```bash
$ ls -lh /opt/android_sdk_tools
total 20K
drwxr-xr-x. 3 ts ts 4.0K Feb 19 11:26 build-tools
drwxr-xr-x. 2 ts ts 4.0K Feb 19 11:22 licenses
drwxr-xr-x. 3 ts ts 4.0K Feb 19 11:28 platforms
drwxr-xr-x. 5 ts ts 4.0K Feb 19 11:22 platform-tools
drwxr-xr-x. 6 ts ts 4.0K Feb 19 11:25 tools
```
